### PR TITLE
new package: movit

### PR DESCRIPTION
**Pull request by knyipab**
Following #20351, try to make a PR here to see if Actions of this repo can build `movit` package which is supposed to require on device build.

### CODE REVIEW

File: packages/movit/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+TERMUX_PKG_SHA256=b33073b705f0ccb6ac4942cf51151515407b40bb4e9a2dd0228c1c2cb1fbc11a
+TERMUX_PKG_DEPENDS="fftw, libepoxy"
+TERMUX_PKG_BUILD_DEPENDS="eigen, googletest, sdl2"
+TERMUX_PKG_SUGGESTS="$TERMUX_PKG_BUILD_DEPENDS"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_AUTO_UPDATE=true
+# build strategy used by Fedora
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--disable-static"
+TERMUX_PKG_EXTRA_MAKE_ARGS="TESTS="
+TERMUX_PKG_BLACKLISTED_ARCHES="arm"
```
Comment (TomJo2000) [inline packages/movit/build.sh:16]: Is the build generally broken on 32bit ARM, or what is the hangup with building it for that architecture?

File: packages/movit/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+TERMUX_PKG_HOMEPAGE=https://movit.sesse.net/
+TERMUX_PKG_DESCRIPTION="The modern video toolkit. "
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_VERSION=1.7.1
+TERMUX_PKG_SRCURL=https://movit.sesse.net/movit-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=b33073b705f0ccb6ac4942cf51151515407b40bb4e9a2dd0228c1c2cb1fbc11a
+TERMUX_PKG_DEPENDS="fftw, libepoxy"
+TERMUX_PKG_BUILD_DEPENDS="eigen, googletest, sdl2"
+TERMUX_PKG_SUGGESTS="$TERMUX_PKG_BUILD_DEPENDS"
```
Comment (TomJo2000) [inline packages/movit/build.sh:10]: Why would the finished package recommend its build time dependencies?

**Comment by TomJo2000** [comment]
Interesting, the builds are actually failing differently between x86 and Aarch64.

File: packages/movit/build.sh (hunk 2 above)
Comment (TomJo2000) [reply packages/movit/build.sh:10]: Also I just noticed, `sdl2` is in the x11 repo, so this package would also need to be in the x11 section.

File: packages/movit/build.sh (hunk 1 above)
Comment (knyipab) [reply packages/movit/build.sh:16]: Below TUR PR Actions prompt error when build 32-bit ARM package (https://github.com/termux-user-repository/tur/actions/runs/9288882338/job/25561402931?pr=1054) when scanning and packaging files. 

Not sure if it is related to build test being disabled (following the practice of [Fedora build script](https://src.fedoraproject.org/rpms/movit/blob/epel8/f/movit.spec)). Archlinux build script relies on more [complete `gtest` installation](https://gitlab.archlinux.org/archlinux/packaging/packages/gtest/-/blob/main/PKGBUILD?ref_type=heads). 

I am not sure, shall I use `TERMUX_PKG_UNDEF_SYMBOLS_FILES` to suppress the error?

```
install -m 0644 effect_chain.h effect_util.h effect.h input.h image_format.h init.h util.h defs.h resource_pool.h fp16.h ycbcr.h version.h flat_input.h ycbcr_input.h ycbcr_422interleaved_input.h lift_gamma_gain_effect.h white_balance_effect.h gamma_expansion_effect.h gamma_compression_effect.h colorspace_conversion_effect.h alpha_multiplication_effect.h alpha_division_effect.h saturation_effect.h blur_effect.h diffusion_effect.h glow_effect.h unsharp_mask_effect.h mix_effect.h overlay_effect.h padding_effect.h resample_effect.h dither_effect.h deconvolution_sharpen_effect.h fft_pass_effect.h vignette_effect.h slice_effect.h complex_modulate_effect.h luma_mix_effect.h fft_convolution_effect.h ycbcr_conversion_effect.h deinterlace_effect.h sandbox_effect.h mirror_effect.h resize_effect.h multiply_effect.h fft_input.h /data/data/com.termux/files/usr/include/movit/
mkdir -p /data/data/com.termux/files/usr/lib/pkgconfig/
install -m 644 movit.pc /data/data/com.termux/files/usr/lib/pkgconfig/
INFO: READELF=llvm-readelf ... /data/data/com.termux/files/usr/bin/llvm-readelf
INFO: Generating symbols regex to /data/data/com.termux/files/usr/tmp/tmp.SOfu4q9HhO
INFO: Done ... 0s
INFO: Total symbols 2136
INFO: Identifying files with nproc=2
INFO: Done ... 1s
INFO: Found 1 / 49 files
INFO: Running symbol checks on 1 files with nproc=2
INFO: Done ... 0s
INFO: Found files with undefined symbols
INFO: Showing result
ERROR: ./lib/libmovit.so contains undefined symbols:
    69: 00000000     0 NOTYPE  GLOBAL DEFAULT   UND __aeabi_uidiv
   120: 00000000     0 NOTYPE  GLOBAL DEFAULT   UND __aeabi_idiv
   155: 00000000     0 NOTYPE  GLOBAL DEFAULT   UND __aeabi_ul2d
   205: 00000000     0 NOTYPE  GLOBAL DEFAULT   UND __aeabi_uidivmod
   217: 00000000     0 NOTYPE  GLOBAL DEFAULT   UND __aeabi_idivmod
INFO: Done ... 1s
ERROR: Refer above
INFO: Found 1 files with undefined symbols after exclusion
Error: Process completed with exit code 1.
```

File: packages/movit/build.sh (hunk 2 above)
Comment (knyipab) [reply packages/movit/build.sh:10]: Should be a mistake of mine when copying from `mlt` build script (which other linux distros usually use similar lists of build deps and optional deps)

File: packages/movit/build.sh (hunk 1 above)
Comment (TomJo2000) [reply packages/movit/build.sh:16]: So it builds, but the symbol checker complains.

**Comment by TomJo2000** [comment]
Please do keep your commits squashed by the way.
You can use `git rebase -i HEAD~2` to do so.
And you will probably want to commit subsequent revisions with `git commit --amend` to amend the HEAD commit.

See https://www.baeldung.com/ops/git-squash-commits#1-squash-the-last-x-commits for for info on interactive rebase.

**Comment by knyipab** [comment]
I rectified the build script following your review. 

After all, the similarity in x86 and aarch64 build is that they both fail to run `./make_bundled_shaders` to generate cpp file, error prompt is a bit different cuz x86 host tries to runs aarch64 binary. 

I pause here and perhaps wait for someone who knows any around to cross-compile it. And also for fixing the symbol issue for `arm` if that's do-able.

**Comment by TomJo2000** [comment]
Regarding the missing symbols on arm.
Code search actually dug up an existing solution.
https://github.com/termux/termux-packages/blob/6cac06b976a669f3a1d713ea2320b9a246e03a7b/packages/marisa/build.sh#L12-L20
May need to be adjusted slightly, but `LDFLAGS+=" $($CC -print-libgcc-file-name)"` seems to be the operative part.

**Comment by knyipab** [comment]
Thanks a lot! The `arm` fix works as I updated in [the TUR PR](https://github.com/termux-user-repository/tur/pull/1054).  

Still, I highly doubt if this package can be cross-compiled. More work may be needed to look into movit's build script.

**Comment by TomJo2000** [comment]
sorry about the delay on the CI runs, I need to approve them manually since this is a first time contribution

**Comment by TomJo2000** [comment]
huh just rewrite it in Python, that's of course also an option.
Nice thinking @licy183.

**Review by TomJo2000 — APPROVED** [review]

**Comment by knyipab** [comment]
thanks a lot @licy183

**Comment by TomJo2000** [comment]
I did also test building shortcut with `movit` from this PR and it built fine.

**Comment by knyipab** [comment]
Tested with shotcut (settings -> GPU effects (unstable)) on my device (Snapdragon 8 Gen 2)
The movit track editor is signjficantly slower than cpu mode except for zink in dri3 + generic vulkan loader. 
The export outputting a overlay video does not work well (only the top video rendered). 
I tried a number of GPU acceleration setups but don't find any useful. 

Still, we may add this for mlt framework and shotcut, provided that it is an opt-in feature.

**Pull request merged**